src/pages/ProductList/ProductList.js

```diff
@@ -1,13 +1,167 @@
-import React, { Component } from 'react'
+import React, { Component } from "react";
+import Product from "./Components/Product"
+import "./ProductList.scss"
 
 class ProductList extends Component {
-    render() {
-        return (
-            <div>
-                
-            </div>
-        )
+  constructor() {
+    super();
+    this.state = {
+      filterList: [],
+      allProducts: [],
+      productsByCategory: [],
+      productsByPage: [],
+      mappingPage: false,
+      isBtnVisible: true,
+      isPageFooterVisible: true,
+      offset: 0
     }
-}
+  }
 
-export default ProductList
+  componentDidMount() {
+    const LIMIT = 12;
+    const APIOfProductFilterList = `http://localhost:3000/Data/productFilterList.json`;
+    const APIOfProductList = `http://localhost:3000/Data/productList.json?limit=${LIMIT}&offset=${this.state.offset}`;
+    // pagination(limit, offset)은 추후 시도 예정입니다.
+
+    Promise.all([
+      fetch(APIOfProductFilterList, {method: "GET"})
+        .then(res => res.json())
+        .then(res => {
+          this.setState({
+            filterList: res.filterList,
+          });
+        }).catch(error => console.log(error.message)),
+
+      fetch(APIOfProductList, {method: "GET"})
+        .then(res => res.json())
+        .then(res => {
+          this.setState({
+            allProducts: res.products,
+            productsByCategory: res.products
+          })
+        })
+    ])
+  }
+
+  filterByCategory = (category) => {
+    if (category === "ALL") {
+      this.setState({
+        productsByCategory: [...this.state.allProducts],
+        mappingPage : false,
+        isBtnVisible: true,
+        isPageFooterVisible: true
+      })
+    } 
+    else {
+      const filtering = this.state.allProducts.filter(product => {
+        return product.category === category;
+      })
+      if (filtering.length < 13) {
+        this.setState({isPageFooterVisible: false})
+      } else {
+        this.setState({isPageFooterVisible: true})
+      }
+      this.setState({
+        productsByCategory: [...filtering],
+        productsByPage: [...filtering]
+      })
+    }
+  }
```

Comment:
```suggestion
  filterByCategory = (category) => {
    if (category === "ALL") {
      return this.setState({
        products: [...this.state.allProducts],
        mappingPage : false,
        isBtnVisible: true,
        isPageFooterVisible: true
      })
    }

    const filteredProducts = this.state.allProducts.filter(product => {
      return product.category === category;
    })

    this.setState({
      isPageFooterVisible: filteredProducts.length >= 13,
      products: [...filteredProducts],
    })
  }
```
- 충분히 고민해보셨을 것 같아서 수정한 코드 제안드립니다
- productsByCategory, productsByPage 두 가지 state에 동일한 값이 들어가는 것 같아 하나로 줄였습니다. 분할할 필요가 있다면 그대로 쓰시면 됩니다.
- `filtering` -> `filteredProducts`으로 변수명 변경했습니다
- 카테고리가 `ALL`일 경우 setState 실행하면서 함수를 리턴하도록 만들었습니다.
- 첫 번째 if문을 통과한 경우에 대해 else문을 작성할 필요가 없습니다.
- setState도 한번 실행될 것이므로 key 값들을 모아서 한번에 처리할 수 있습니다.

Reply:
리뷰 정말 감사합니다 🙇🏻‍♀️

* filterByPage 함수에서 productsByCategory를 slice한 값을 productsByPage state에 setState하기 때문에 둘은 그대로 분할하여 사용하겠습니다
* 금요일에 지영 멘토님 리뷰 반영한 후 재 push 했습니다! 수정된 파일에서는 종택님께서 피드백 주신 것과 동일하게 `filtering` -> `filteredProducts`으로 변수명을 변경했어요
* `return this.setState({코드코드})` 식으로 작성하는 이유는, setState 실행 후 해당 함수가 바로 종료되도록 하기 위해서인가요? 
* else 부분을 다시 한 번 삭제해봤더니, 이전과 동일하게 작동하네요! 😃 감사합니다 ㅎㅎ
* 우선 아래 2가지 경우대로 수정해봤습니다. 줄 수가 더 짧은 첫 번째 경우가 더 나은가 싶다가도, setState를 최소화 한 두 번째 경우가 더 나은 것 같기도 한데, 종택님은 어느 것이 더 효율적으로 보이시나요?
~~~
this.setState({
        productsByCategory: [...filteredProducts],
        productsByPage: [...filteredProducts]
      })
      this.setState(filteredProducts.length < 13 ? {isPageFooterVisible: false} : {isPageFooterVisible: true})
~~~
~~~
this.setState(filteredProducts.length < 13 ? 
        {isPageFooterVisible: false,
        productsByCategory: [...filteredProducts],
        productsByPage: [...filteredProducts]} 
        : 
        {isPageFooterVisible: true, 
        productsByCategory: [...filteredProducts],
        productsByPage: [...filteredProducts]}
      )
~~~

Reply:
```js
this.setState({
  isPageFooterVisible: filteredProducts.length < 13,
  productsByCategory: [...filteredProducts],
  productsByPage: [...filteredProducts]} 
})
```
앞선 댓글에서 보여드렸던 대로 이 방식이 좋을 것 같습니다

Reply:
아! `filteredProducts.length < 13` 자체가 boolean 값을 내기 때문에 그걸 이용한 거군요!!

코드블럭 꿀팁 감사합니다 ㅎㅎ